File: finalRest/src/main/java/ajc/formation/project/finalRest/entity/TypeReseau.java
package ajc.formation.alten.finalRest.entity;

import java.sql.Date;

import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonView;

import ajc.formation.alten.finalRest.jsonview.Views;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinColumns;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "TYPERESEAU")
@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class TypeReseau  {

	@JsonView(Views.Common.class)
	@Id
	@EqualsAndHashCode.Include
	@Column(name = "IDTYPERESEAU")
	@NotNull
	private Long idTypeReseau;

	@JsonView(Views.Common.class)
	@Column(name = "LIBELLE", length = 60)
	private String libelle;

	@DateTimeFormat(pattern = "yyyy-MM-dd")
	@Column(name = "DATEDERNIEREMODIFICATION")
	private Date dateDerniereModification;

	@Column(name = "REJET")
	private Integer rejet;
	
	@ManyToOne
	@JoinColumns({
		@JoinColumn(name = "IDRESEAU",referencedColumnName="IDRESEAU", foreignKey = @ForeignKey(name = "I_FK_TYPERESEAU_RESEAU")),
		@JoinColumn(name = "IDCODEINTERNERESEAU", referencedColumnName="IDCODEINTERNERESEAU",foreignKey = @ForeignKey(name = "I_FK_TYPERESEAU_RESEAU"))
	})
	private Reseau reseau;
}

File: finalAng/src/app/model/type-reseau.ts
import { Reseau } from './reseau';

export class TypeReseau {
  public get reseau(): Reseau | undefined {
    return this._reseau;
  }
  public set reseau(value: Reseau | undefined) {
    this._reseau = value;
  }
  public get rejet(): number | undefined {
    return this._rejet;
  }
  public set rejet(value: number | undefined) {
    this._rejet = value;
  }
  public get dateDerniereModification(): Date | undefined {
    return this._dateDerniereModification;
  }
  public set dateDerniereModification(value: Date | undefined) {
    this._dateDerniereModification = value;
  }
  public get libelle(): string | undefined {
    return this._libelle;
  }
  public set libelle(value: string | undefined) {
    this._libelle = value;
  }
  public get idTypeReseau(): number | undefined {
    return this._idTypeReseau;
  }
  public set idTypeReseau(value: number | undefined) {
    this._idTypeReseau = value;
  }
  constructor(
    private _idTypeReseau?: number,
    private _libelle?: string,
    private _dateDerniereModification?: Date,
    private _rejet?: number,
    private _reseau?: Reseau
  ) {}
}
